src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Link } from 'react-router-dom';
+
+export enum StatusOfFilter {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
+
+type Props = {
+  setFilter: React.Dispatch<React.SetStateAction<StatusOfFilter>>;
+  filter: StatusOfFilter;
+};
+
+export const TodoFilter: React.FC<Props> = ({ filter, setFilter }) => {
+  return (
+    <ul className="filters">
+      <li>
+        <Link
+          to="#/"
+          className={classNames({ selected: filter === StatusOfFilter.All })}
+          onClick={() => setFilter(StatusOfFilter.All)}
```

Comment:
Good practice is create separate handler instead of using anonymous functions
Try to fix all similar places